mongo/src/main/java/com/redhat/lightblue/mongo/metadata/BSONParser.java

```diff
@@ -132,14 +126,52 @@ public Object asRepresentation(Object value) {
     }
 
     /**
-     * Override to set _id appropriately.
+     * Override to convert partialFilterExpression index property from string to map.
+     *
+     */
+    @Override
+    public EntityInfo parseEntityInfo(Object object) {
+
+        List<DBObject> indexes = (List<DBObject>) getMapProperty(object, "indexes");
+
+        if (indexes != null) {
+            for (DBObject index: indexes) {
+                String partialFilterExpression = (String) index.get(MongoCRUDController.PARTIAL_FILTER_EXPRESSION_OPTION_NAME);
+
+                if (partialFilterExpression != null) {
+                    // convert string to json
+                    // https://github.com/lightblue-platform/lightblue-mongo/issues/329
+                    index.put(MongoCRUDController.PARTIAL_FILTER_EXPRESSION_OPTION_NAME, (Map<String,Object>)JSON.parse(partialFilterExpression));
+                }
+            }
+        }
+
+        return super.parseEntityInfo(object);
+    }
```

Comment:
Ideally, you should convert it back to string after you're done with it, in case the BSON object is used somewhere else. It is invalid at this point.